lib/tbot/tbot.go

```diff
@@ -143,40 +143,49 @@ func (b *Bot) initialize(ctx context.Context) error {
 		return trace.Wrap(err, "could not read bot storage destination from config")
 	}
 
-	configTokenHashBytes := []byte{}
-	if b.cfg.Onboarding != nil && b.cfg.Onboarding.Token != "" {
-		sha := sha256.Sum256([]byte(b.cfg.Onboarding.Token))
-		configTokenHashBytes = []byte(hex.EncodeToString(sha[:]))
-	}
-
 	var authClient auth.ClientI
 
+	fetchNewIdentity := true
 	// First, attempt to load an identity from storage.
 	ident, err := identity.LoadIdentity(dest, identity.BotKinds()...)
-	if err == nil && !hasTokenChanged(ident.TokenHashBytes, configTokenHashBytes) {
-		identStr, err := describeTLSIdentity(ident)
-		if err != nil {
-			return trace.Wrap(err)
+	if err == nil {
+		if b.cfg.Onboarding != nil && b.cfg.Onboarding.HasToken() {
+			// try to grab the token to see if it's changed, as we'll need to fetch a new identity if it has
+			if token, err := b.cfg.Onboarding.Token(); err == nil {
+				sha := sha256.Sum256([]byte(token))
+				configTokenHashBytes := []byte(hex.EncodeToString(sha[:]))
+
+				fetchNewIdentity = hasTokenChanged(ident.TokenHashBytes, configTokenHashBytes)
+			} else {
+				// we failed to get the token, we'll continue on trying to use the existing identity
+				fetchNewIdentity = false
```

Comment:
I think we need to at least log the error from reading the token file here, this could occur if the user the bot is running as changes or the permissions of the token file change. The log message should probably also provide an explanation that the last good identity will be used.

Reply:
Good call!

Reply:
I wasn't sure if we should include the error and the notice that we'll use the last good identity in the same message, so I did the error as `log.Errorf` and then another `log.Info` saying that we'll use the last good identity. Let me know if you had something else in mind!